### PR TITLE
Update EIP-1: Clear rules regarding EIP numbering

### PR DESCRIPTION
**Pull request by Pandapip1**
This also paves the way for "draft" EIP numbers being officially used.

### CODE REVIEW

File: EIPS/eip-1.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ The following is the standardization process for all EIPs in all tracks:
 
 **Idea** - An idea that is pre-draft. This is not tracked within the EIP Repository.
 
-**Draft** - The first formally tracked stage of an EIP in development. An EIP is merged by an EIP Editor into the EIP repository when properly formatted.
+**Draft** - The first formally tracked stage of an EIP in development. An EIP is merged by an EIP Editor into the EIP repository when properly formatted. EIP editors and accounts authorized by any EIP editor can assign EIP numbers, which need not be unsigned integers. EIP numbers can also be unassigned by the account that assigned that number, as long as it is not the only assigned number. If multiple numbers are assigned, the EIP author is free to choose from any of the assigned numbers.
```
[the remark this answers is not in the document]
Comment (abcoathup) [reply EIPS/eip-1.md:78]: Is there more info on how draft numbers work?

The current process as I understand it is: 
EIP editor assigns the PR number as the EIP/ERC number unless there were previous  issues/PRs then the earliest issue/PR number should be used, or there is number gaming or lunging (e.g. PR with no spec or little details), then a non-EIP related PR number below the PR number can be issued.

[the remark this answers is not in the document]
Comment (Pandapip1) [reply EIPS/eip-1.md:78]: When I refer to "draft" EIP numbers, I'm referring to 'numbers' like `draft_proposal_name_here` (for an identifier of `EIP-draft_proposal_name_here`).

As for the process of assigning numbers, it is soon to be automated.

[the remark this answers is not in the document]
Comment (abcoathup) [reply EIPS/eip-1.md:78]: > "draft" EIP numbers

Non-numeric EIP/ERC numbers are harder to succinctly refer to.  e.g. in core devs discussions, or in newsletters

**Comment by timbeiko** [comment]
Sorry if I missed part of the conversation here, but what is the rationale to not restrict numbering to unsigned integers? 

Many EIPs end up in draft for far too long, and having the number change can cause confusion both within the community and potentially some technical issues when it changes in clients, along with the extra work for clients to have to support alphanumeric characters in their configs, as mentioned by @g11tech. 

I'm also mildly against re-using old numbers: we've got about 1/10 used in the current scheme, why not simply [backfill to a never used one](https://github.com/ethereum/EIPs/issues/6990)?

**Comment by SamWilsn** [comment]
> Sorry if I missed part of the conversation here, but what is the rationale to not restrict numbering to unsigned integers?

In pre-draft pull requests, the EIP's "number" is an escaped version of the title. Some editors would like to extend those non-numbered file names into draft proposals. It makes it much clearer that the proposal is still a draft, and shouldn't be depended upon yet.

> Many EIPs end up in draft for far too long

That is part of the motivation to make drafts even draftier.

> having the number change can cause confusion both within the community and potentially some technical issues when it changes in clients

There would be no number change. You'd go from `eip-this_is_my_title.md` to `eip-1234.md` when you go from `Draft` to `Review`, hypothetically.

**Comment by timbeiko** [comment]
Fair enough, then re: "That is part of the motivation to make drafts even draftier.", could it be possible to move from Draft to Review without requiring an Editor +1?

**Comment by abcoathup** [comment]
> There would be no number change. You'd go from `eip-this_is_my_title.md` to `eip-1234.md` when you go from `Draft` to `Review`, hypothetically.

Long draft names will be a challenge for those referring to them.  Especially when titles change.
I would much prefer a prefix or suffix to indicate draftiness DRAFT-EIP-9999, which is then dropped at the review stage.

How does this scheme work with auto-assigned EIP numbers?

**Comment by lightclient** [comment]
i also prefer referring to draft EIPs with a prefix instead of long draft names

File: EIPS/eip-1.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pandapip1) [reply EIPS/eip-1.md:78]: > Non-numeric EIP/ERC numbers are harder to succinctly refer to. e.g. in core devs discussions, or in newsletters

I'm hoping that EIPs shouldn't need to stay in draft for long.

[the remark this answers is not in the document]
Comment (Pandapip1) [reply EIPS/eip-1.md:78]: That makes sense.

[the remark this answers is not in the document]
Comment (Pandapip1) [reply EIPS/eip-1.md:78]:
```suggestion
**Draft** - The first formally tracked stage of an EIP in development. An EIP is merged by an EIP Editor into the EIP repository when properly formatted. EIP editors and accounts authorized by any EIP editor can assign EIP numbers, which need not be unsigned integers for EIPs with a status of Draft. EIP numbers can also be unassigned by the account that assigned that number, as long as it is not the only assigned number. If multiple numbers are assigned, the EIP author is free to choose from any of the assigned numbers.
```

**Comment by abcoathup** [comment]
> The concern that was brought up was that if actual numbers were assigned at Draft, then there would be a lot of spam if #5641 were implemented.

Assuming automerging of PRs once they pass validation is a spam risk, and EIP numbers could be wasted if they were autoassigned at this stage:  Instead we need a quick mechanism for EIP editors to assign automatic EIP numbers, such as a tag or a label once an EIP has been merged.  EIP numbers could be increasing with some random element to avoid any perception of favored number assignment.

File: EIPS/eip-1.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lightclient) [reply EIPS/eip-1.md]: Unfortunately they tend to.

**Comment by Pandapip1** [comment]
> Assuming automerging of PRs once they pass validation is a spam risk, and EIP numbers could be wasted if they were autoassigned at this stage: Instead we need a quick mechanism for EIP editors to assign automatic EIP numbers, such as a tag or a label once an EIP has been merged.

That's what the `draft_*` EIP numbers are meant to solve.

> EIP numbers could be increasing with some random element to avoid any perception of favored number assignment.

That's what eip-review-bot is going to do once some of the issues are sorted out.

**Review by Scamreno — COMMENTED** [review]
I'm just trying to get this process taken care of because like I was just trying to find a way to just to get some of my funds to my account because right now I got I have a lot of assets and I know what's going on but I don't understand why I'm getting it off so I've been taking me a while right now me and my family homeless right now so if there's anybody on this network they can be able to help me out so I can believe it get these things taken care of please let me know

File: EIPS/eip-1.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ The following is the standardization process for all EIPs in all tracks:
 
 **Idea** - An idea that is pre-draft. This is not tracked within the EIP Repository.
 
-**Draft** - The first formally tracked stage of an EIP in development. An EIP is merged by an EIP Editor into the EIP repository when properly formatted. EIP editors and accounts authorized by any EIP editor can assign EIP numbers, which need not be unsigned integers. EIP numbers can also be unassigned by the account that assigned that number, as long as it is not the only assigned number. If multiple numbers are assigned, the EIP author is free to choose from any of the assigned numbers.
+**Draft** - The first formally tracked stage of an EIP in development. An EIP is merged by an EIP Editor into the EIP repository when properly formatted. EIP editors and accounts authorized by any EIP editor can assign EIP numbers, which need not be unsigned integers for EIPs with a status of Draft. EIP numbers can also be unassigned by the account that assigned that number, as long as it is not the only assigned number. If multiple numbers are assigned, the EIP author is free to choose from any of the assigned numbers.
```
Comment (Scamreno) [inline EIPS/eip-1.md:78]: Only

**Review by Scamreno — COMMENTED** [review]
@Pandapip1

**Comment by Scamreno** [comment]
Uploading   @@VID_20240618_183251.mp4…7

File: EIPS/eip-1.md (hunk 2 above)
Comment (Scamreno) [reply EIPS/eip-1.md:78]: I am ready for my review

**Review by Scamreno — COMMENTED** [review]
#6976

File: EIPS/eip-1.md (hunk 2 above)
Comment (Scamreno) [reply EIPS/eip-1.md:78]: I also was trying to open some of my old files that because of non-inactivity #8630 #8682 #8369 #6976

Comment (Tsukimarf) [reply EIPS/eip-1.md:78]: Please my review

**Pull request closed without merging**